### ISSUE TITLE
Song select freezes upon opening and interacting with large collections.

### ISSUE THREAD

**Issue by Wrongwayscout** [ppy/osu#26388]
### Type

Performance

### Bug description

I have an unnecessarily large collection (~30k difficulties) and another collection about a tenth the size, both experience freezes upon opening them.
The larger collection freezes for around 20 seconds, and the ~3,000 difficulty one freezes for about 2.
The freeze also occurs when exiting out of a search within these collections with esc, again about 20 seconds for the larger one and 2 for the smaller one.

These collections both perform very well on stable (slight framedrops when first scrolling through).

### Screenshots or videos

Opening a collection with 29,042 difficulties:
https://github.com/ppy/osu/assets/84840821/ed1caafb-913e-40af-98a0-837111b5b55a

Exiting a search within a collection with 29,042 difficulties:
https://github.com/ppy/osu/assets/84840821/deb22a90-302f-4861-9a95-d790fcb3bd23

Opening a collection with 2,803 difficulties:
https://github.com/ppy/osu/assets/84840821/3235a092-9ad2-46c2-8f81-12f6e9c686ab

Exiting a search within a collection with 2,803 difficulties:
https://github.com/ppy/osu/assets/84840821/c07a6172-3df5-4197-8522-b48bffe91133

### Version

2023.1231.0

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/13830503/compressed-logs.zip)

**Comment by hachuyenden** [comment]
This is true, because the collection is imported from stable to lazer. Selecting that collection in the lazer will be much slower (3 times) compared to a collection created in the lazer itself.
